sdcm/utils/operator/multitenant_common.py

```diff
@@ -0,0 +1,148 @@
+        self._es_doc_type = "test_stats"
+        self._stats = self._init_stats()
+        self.test_config = test_config
+        self._duration = self.params.get(key='test_duration')
+        self.create_stats = self.params.get(key='store_perf_results')
+        self.status = "RUNNING"
+        self.cluster_index = str(cluster_index)
+        self._test_id = self.test_config.test_id() + f"--{cluster_index}"
+        self._test_index = self.get_str_index()
+        self.create_test_stats()
+        self.start_time = self.get_test_start_time() or time.time()
+        self.timeout_thread = self._init_test_timeout_thread()
+        self.test_config.reuse_cluster(False)
+
+    def get_str_index(self):
+        return f"{self.get_str_index_prefix}-{self.db_cluster.k8s_cluster.tenants_number}-tenants"
+
+    def id(self):  # pylint: disable=invalid-name
+        if "Longevity" in self.__class__.__name__:
+            return f"{self.test_config.test_id()}-{self._test_index}"
+        # NOTE: performance results should not have unique IDs to be able to be used in comparisons
+        return f"id--{self._test_index}"
+
+    def __str__(self) -> str:
+        return self._test_index + f"--{self.cluster_index}"
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+
+def get_tenants(test_class_instance):  # pylint: disable=too-many-branches,too-many-locals
+    parent_test_class = None
+    for base in test_class_instance.__class__.__bases__:
+        if base.__name__.endswith("Test"):
+            parent_test_class = base
+            break
+    else:
+        test_class_instance.log.warning(
+            "Could not find parent test class. Tenant classes won't inherit anything.")
+    tenant_class_name, get_str_index_prefix = f"TenantFor{parent_test_class.__name__}", "k8s"
+    if "Performance" in tenant_class_name:
+        get_str_index_prefix += "-perf"
+    elif "Longevity" in tenant_class_name:
+        get_str_index_prefix += "-longevity"
+
+    # Init tenants
+    tenants = []
+    for i, db_cluster in enumerate(test_class_instance.db_clusters_multitenant):
+        current_cluster_index = i + 1
+        current_tenant_class = type(
+            f"{tenant_class_name}-{current_cluster_index}", (TenantMixin, parent_test_class), {
+                "get_str_index_prefix": get_str_index_prefix,
+            }
+        )
+        tenants.append(current_tenant_class(
+            db_cluster=db_cluster,
+            loaders=test_class_instance.loaders_multitenant[i],
+            monitors=test_class_instance.monitors_multitenant[i],
+            prometheus_db=test_class_instance.prometheus_db_multitenant[i],
+            params=test_class_instance.params,
+            test_config=test_class_instance.test_config,
+            cluster_index=current_cluster_index,
+        ))
+
+    # Process multitenant parameters if present
+    for param_dict in test_class_instance.params.config_options:
```

Comment:
shouldn't this be part of the configuration object ? i.e. just tell it the index, and it would return the correct configuration object ?